src/main/java/ca/uhn/fhir/jpa/starter/dotbase/api/FhirServer.java

```
@@ -0,0 +1,56 @@
+package ca.uhn.fhir.jpa.starter.dotbase.api;
+
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpClient.Redirect;
+import java.net.http.HttpClient.Version;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+
+public class FhirServer {
+  private static final org.slf4j.Logger ourLog = org.slf4j.LoggerFactory.getLogger(
+    FhirServer.class
+  );
+  private static final HttpClient HTTP_CLIENT = FhirServer.clientConfig();
+
+  private static final HttpClient clientConfig() {
+    return HttpClient
+      .newBuilder()
+      .version(Version.HTTP_2)
+      .connectTimeout(Duration.ofSeconds(30))
+      .followRedirects(Redirect.NEVER)
+      .build();
+  }
+
+  private static String getAuthHeader(RequestDetails theRequestDetails) {
+    if (theRequestDetails.getHeader("Authorization") != null) {
+      return theRequestDetails.getHeader("Authorization");
+    }
+    return "";
```

Shouldn't this throw an error instead of returning an empty string?

---

I would stick to not throwing an Error here, since it depends on the FHIR server that you request, if Authorization is necessary or not. E.g. requests to public servers would also fail here otherwise